astropy/units/core.py

```diff
@@ -447,6 +481,26 @@ def add_enabled_equivalencies(equivalencies):
     return context
 
 
+def unit_aliases(aliases):
+    """
+    Context manager for temporarily adding aliases for units (e.g. to handle
+    alternate spellings).
+
+    This should be preferred over creating a new unit and manually
+    converting to the correct unit.
+
+    Parameters
+    ----------
+    aliases : dict (or other mapping) of str:unit pairs
+        The aliases to temporarily use. The keys must be the aliases stored
+        as strings, which must not have been used for any existing units (a
+        ValueError is raised otherwise). The values must be the astropy unit
+        that the alias will be mapped to.
+    """
+    registry = get_current_unit_registry()
```

Comment:
So, this would become:
```
    # get a context with a new registry, which is a copy of the current one
    context = _UnitContext(get_current_unit_registry())
    # in this new current registry, enable the further units requested
    get_current_unit_registry().add_aliased_units(aliases)
    return context
```